site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md

```diff
@@ -0,0 +1,469 @@
+}
+```
+
+The following code save the default suggestions to storage when the extension is first installed:
+
+{% Label %}sw-omnibox.js:{% endLabel %}
+
+```js
+...
+// Save default API suggestions
+chrome.runtime.onInstalled.addListener(({ reason }) => {
+  if (reason === 'install') {
+    chrome.storage.local.set({
+      apiSuggestions: ['tabs', 'storage', 'scripting']
+    });
+  }
+});
+```
+
+Service workers do not have direct access to the [window object][mdn-window], therefore cannot use
+[window.localStorage()][mdn-local-storage] to store values. Also, service workers are short-lived execution environments;
+they get terminated repeatedly throughout a user's browser session, which makes it incompatible with
+global variables.
+
+See [Saving state](TBD) to learn about storage options for extension service workers.
+
+### Step 5: Register your events {: #step-5 }
+
+All event listeners need to be registered in the global scope of the service worker. In other words, event listeners should not be nested in functions. This way Chrome can immediately invoke all event handlers, even if the extension's async startup logic hasn't finished. 
+
+To use the [`chrome.omnibox`][api-omnibox] API first add the omnibox keyword to the manifest:
+
+{% Label %}manifest.json:{% endLabel %}
+
+```json
+{
+  ...
+  "minimum_chrome_version": "102",
+  "omnibox": {
+    "keyword": "api"
+  },
+}
+```
+
+{% Aside 'important' %}
+The [`"minimum_chrome_version"`][manifest-min-version] explains how this key behaves when a user tries to install your extension but isn't using a compatible version of Chrome.
+
+{% endAside %}
+
+The following code registers the omnibox event listeners at the top level of the script and updates [chrome.storage][api-storage] with the most recent api search.
+
+{% Label %}sw-omnibox.js:{% endLabel %}
+
+```js
+...
+const URL_CHROME_EXTENSIONS_DOC =
+  'https://developer.chrome.com/docs/extensions/reference/';
+const NUMBER_OF_PREVIOUS_SEARCHES = 4;
+
+// Display the suggestions after user starts typing
+chrome.omnibox.onInputChanged.addListener(async (input, suggest) => {
+  await chrome.omnibox.setDefaultSuggestion({
+    description: 'Enter a Chrome API or choose from past searches'
+  });
+  const { apiSuggestions } = await chrome.storage.local.get('apiSuggestions');
+  const suggestions = apiSuggestions.map((api) => {
+    return { content: api, description: `Open chrome.${api} API` };
+  });
+  suggest(suggestions);
+});
+
+// Open the reference page of the chosen API
+chrome.omnibox.onInputEntered.addListener((input) => {
+  chrome.tabs.create({ url: URL_CHROME_EXTENSIONS_DOC + input });
+  // Save the latest keyword
+  updateHistory(input);
+});
+
+async function updateHistory(input) {
+  const { apiSuggestions } = await chrome.storage.local.get('apiSuggestions');
+  apiSuggestions.unshift(input);
+  apiSuggestions.splice(NUMBER_OF_PREVIOUS_SEARCHES);
+  await chrome.storage.local.set({ apiSuggestions });
```

Comment:
Easier way (I think):

```
const { apiSuggestions } = await chrome.storage.local.get('apiSuggestions');
apiSuggestions = input.concat(apiSuggestions);
await chrome.storage.local.set({ apiSuggestions });
```

Reply:
concat behaves funny 🙃
<img width="451" alt="Screenshot 2023-03-22 at 09 19 31" src="https://user-images.githubusercontent.com/37001393/226934070-68e1ce62-ec20-4072-b456-0bd5177ce30a.png">

Reply:
Sorry, should've been `[input].concat`. Thinking. more about this, I would ignore the number of suggestions and just go with:

```
const { apiSuggestions } = await chrome.storage.local.get('apiSuggestions');
await chrome.storage.local.set({ [input].concat(apiSuggestions) });
```

But your call :-)

Reply:
Okie dokies 😁